api/models/cartDao.js

```diff
@@ -0,0 +1,99 @@
+const dataSource = require('./dataSource')
+const queryRunner = dataSource.createQueryRunner()
+
+
+const createCart = async ( productId, userId, quantity) => {
+try {
+const result = await dataSource.query(`
+    INSERT INTO carts (
+      user_id,
+      product_id,
+      quantity,
+      status_id
+      ) 
+    VALUES (
+      ?,
+      ?,
+      ?,
+      1
+      )`,
+    [ userId, productId, quantity ]
+    )
+
+    return result.insertId
+    } catch{
+      const error = new Error('cannot create cart')
+      error.statusCode = 400
+      throw error
+    }
+}
+
+const updateCart = async (quantity, productId, userId) =>{
+  await queryRunner.connect()
+  await queryRunner.startTransaction()
+    
+  const updateRows = (await queryRunner.query(`
+        UPDATE carts
+        SET quantity =?
+            product_id = ?
+        WHERE user_id =?`,
+            [quantity ,userId, productId]
+      )).affectedRows
+
+      if (updateRows !== 1) 
+      throw new Error ('UNEXPECTED_NUMBER_UPDATED')
+
+      const result = await queryRunner.query(`
+          SELECT
+          c.user_id,
+          c.product_id,
+          c.quantity,
+          c.status_id,
+          p.thumbnail_image_url,
+          p.price,
+          p.name,
+          u.point
+        FROM carts c
+        INNER JOIN products p ON p.id = c.product_id
+        INNER JOIN users u ON u.id= c.user_id
+        WHERE user_id = ?`, [userId]
+        )
+
+        return result[0]
+      }
```

Comment:
transanction이 성공적으로 실행되었을 경우와 실패했을 경우의 로직을 처리해주어야 할 것 같습니다.

다음과 같이, async - await 키워드를 사용하였을 때 try - catch 구문으로 사용할 수 있습니다.

```suggestion
  try {
    const updateRows = (await queryRunner.query(`
        UPDATE carts
        SET quantity =?
            product_id = ?
        WHERE user_id =?`,
            [quantity ,userId, productId]
      )).affectedRows

      if (updateRows !== 1) 
      throw new Error ('UNEXPECTED_NUMBER_UPDATED')

      const result = await queryRunner.query(`
          SELECT
          c.user_id,
          c.product_id,
          c.quantity,
          c.status_id,
          p.thumbnail_image_url,
          p.price,
          p.name,
          u.point
        FROM carts c
        INNER JOIN products p ON p.id = c.product_id
        INNER JOIN users u ON u.id= c.user_id
        WHERE user_id = ?`, [userId]
        )
        await queryrunner.commitTransaction();
        return result[0]
    } catch (err) {
      await transaction.rollbackTransaction();
    }  finally {
       await transaction.release();
    }
```